### PR TITLE
Caution flag filters backend

### PR DESCRIPTION
**Pull request by ksantiagoBAH**
## Description
As a Comparison Tool user, I need to be able to exclude institutions with caution flags so that I can only look at schools without warnings in the search results.

front end changes: https://github.com/department-of-veterans-affairs/vets-website/pull/12107
database migration pr:
https://github.com/department-of-veterans-affairs/gibct-data-service/pull/636
story:
https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/7185
## Testing done
1. run database migration from PR .
2. seed your database.
3. open up the rails console type `rails console` in gids terminal.
4. run this in the rails console `Institution.reset_counters(institutions.id, :caution_flags)}` to populate the `count_of_caution_flags` column.
5. There are currently no vettec schools with caution flags so update the `count_of_caution_flags` column in institutions in your database, select any vettec school. 
6. Toggle filters

## Screenshots
Screen Shot 2020-04-03 at 8 36 55 AM
Screen Shot 2020-04-03 at 8 37 05 AM

Screen Shot 2020-04-02 at 8 14 01 AM
VETTEC
Screen Shot 2020-04-03 at 8 49 44 AM
Screen Shot 2020-04-03 at 8 49 54 AM

Screen Shot 2020-04-02 at 8 12 38 AM

## Acceptance criteria
1 . A filter is available on the search results screen that meets the following criteria:
- [x] It is available on the traditional search results page.
- [x] It is available on the VET TEC search results page.
- [x] When selected, it removes institutions with caution flags from the search results page
- [x] When de-selected, it displays institutions with caution flags on the search results page that meet all other search criteria.
- [x] It has a Learn More Link to display additional content to the user.
When clicking on the Learn more link (1e), the following information is displayed:

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by sonntag-adam — APPROVED** [review]

File: spec/factories/institutions.rb [hunk 1]
```diff
@@ -111,6 +125,13 @@
       physical_country { 'USA' }
     end
 
+    trait :count_of_caution_flags do
```
Comment (jholton) [inline spec/factories/institutions.rb:128]: This trait is defined twice (see line 101)

File: spec/controllers/v0/institutions_controller_spec.rb [hunk 2]
```diff
@@ -256,6 +256,14 @@ def preview_body(body)
       expect(response.content_type).to eq('application/json')
       expect(response).to match_response_schema('autocomplete')
     end
+
+    it 'filter by caution flags' do
+      create(:institution, :count_of_caution_flags, :start_like_harv)
+      get(:autocomplete, params: { term: 'harv' })
+      expect(JSON.parse(response.body)['data'].count).to eq(0)
```
Comment (jholton) [inline spec/controllers/v0/institutions_controller_spec.rb:263]: You didn't set `exclude_caution_flags` in the `params`.  Shouldn't this then return the institution?

Comment (ksantiagomorgan6) [reply spec/controllers/v0/institutions_controller_spec.rb]: Updated it

File: spec/factories/institution_programs.rb [hunk 3]
```diff
@@ -46,6 +46,10 @@
       institution { create(:institution, preferred_provider: true, version_id: Version.last.id) }
     end
 
+    trait :exclude_caution_flags do
+      institution { create(:institution, exclude_caution_flags: true, version_id: Version.last.id) }
```
[the remark this answers is not in the document]
Comment (ksantiagomorgan6) [reply spec/factories/institution_programs.rb]: I wasn't using that trait, i went ahead and deleted it

File: spec/factories/institutions.rb [hunk 4]
```diff
@@ -92,6 +92,15 @@
       vet_tec_provider { true }
     end
 
+    trait :exclude_caution_flags do
+      institution { 'COLLEGE OF VET TEC PROVIDER' }
+      city { 'CHARLESTON' }
+      state { 'SC' }
+      country { 'USA' }
+      vet_tec_provider { false }
```
[the remark this answers is not in the document]
Comment (ksantiagomorgan6) [reply spec/factories/institutions.rb]: removed unnecessary attributes

File: spec/controllers/v0/institutions_controller_spec.rb [hunk 5]
```diff
@@ -107,6 +107,14 @@ def preview_body(body)
       expect(response).to match_response_schema('autocomplete')
     end
 
+    it 'excludes institutions with caution flags' do
+      create(:institution, :exclude_caution_flags, :start_like_harv)
+      get(:autocomplete, params: { term: 'harv', exclude_caution_flags: true })
+      expect(JSON.parse(response.body)['data'].count).to eq(0)
```
[the remark this answers is not in the document]
Comment (ksantiagomorgan6) [reply spec/controllers/v0/institutions_controller_spec.rb]: Updated spec creating 2 institutions

**Review by jholton — APPROVED** [review]

**Pull request merged**